lib/ldap_fluff/posix_member_service.rb

```diff
@@ -18,10 +18,12 @@ def find_user(uid, base_dn = @base)
   def find_user_groups(uid)
     groups = []
     @ldap.search(
-      :filter => Net::LDAP::Filter.eq('memberuid', uid),
+      :filter => Net::LDAP::Filter.eq(@attr_login, uid),
       :base => @group_base, :attributes => ["cn"]
     ).each do |entry|
-      groups << entry[:cn][0]
+      entry[:memberof].each do |grp|
+        groups << grp.sub(/.*?cn=(.*?),.*/, '\1')
```

Comment:
The regexp you used here assumes there is more than one `key=value` element and I'm not sure if that's always there. Using something like `/.*cn=([^,]+).*/` feels like a safer bet.

Reply:
`#sed` returns the string as-is if it does not match. I'm not sure if it can happen, but again, something like this would probably be safer
```suggestion
        group = grp.match(/cn=([^,]+)/)&.captures&.first
        groups << group if group
```

Reply:
Isn't something like https://stackoverflow.com/questions/50285542/parse-ldap-dn-string-into-key-value-pairs-in-ruby/57413350#57413350 better?

Reply:
That would indeed be better

Reply:
So add `require 'net/ldap/dn'` (after `net/ldap`, before it somehow fails for me) and then
```suggestion
        dn = Net::LDAP::DN.new(grp)
        groups << dn.enum_for(:each_pair).find { |key, _value| key == 'cn' }[1]
```

Now I do wonder if this works though. In the filter you say you only want the `cn` attribute, not `memberof` so I think it won't retrieve the property and `entry[:memberof]` will be `nil`. And even if you retrieve it, it may be `nil` as well for users without `memberof` set.

So you do need to make it a safe navigator iteration: `entry[:memberof]&.each`